tests/tpostgresunique.nim

```diff
@@ -0,0 +1,55 @@
+import unittest
+import strutils
+
+import norm/[model, postgres]
+
+import models
+
+
+const
+  dbHost = "postgres"
+  dbUser = "postgres"
+  dbPassword = "postgres"
+  dbDatabase = "postgres"
+
+proc duplicateInsert(dbConn: DbConn)
+
+suite "Table creation":
+  proc resetDb =
+    let dbConn = open(dbHost, dbUser, dbPassword, "template1")
+    dbConn.exec(sql "DROP DATABASE IF EXISTS $#" % dbDatabase)
+    dbConn.exec(sql "CREATE DATABASE $#" % dbDatabase)
+    close dbConn
+
+  setup:
+    resetDb()
+    let dbConn = open(dbHost, dbUser, dbPassword, dbDatabase)
+
+  teardown:
+    close dbConn
+    resetDb()
+
+  test "Duplicate insert":
+    # Check test fail with DbError
+    expect DbError:
+      duplicateInsert(dbConn)
+
+proc duplicateInsert(dbConn: DbConn)=
```

Comment:
I think this is way overcomplicated. You could just have all that as part of the test, inside the test body.